src-cpp/rdkafkacpp.h

```diff
@@ -1449,15 +1436,14 @@ class RD_EXPORT KafkaConsumer : public virtual Handle {
    */
   virtual ErrorCode position (std::vector<TopicPartition*> &partitions) = 0;
 
-        /**
-         * @brief Retrieve queue for a given partition.
-         *
-         * @returns RD_KAFKA_RESP_ERR_NO_ERROR on success in which case \p queue
-         *          is set to the fetch queue for the given partition. Else,
-         *          return an error code.
-         */
-        virtual ErrorCode partitionQueue (TopicPartition *partition,
-                                          Queue *queue) = 0;
+  /**
+   * @brief Retrieve queue for a given partition.
+   *
+   * @returns RD_KAFKA_RESP_ERR_NO_ERROR on success in which case \p queue
+   *          is set to the fetch queue for the given partition. Else,
+   *          return an error code.
+   */
+  virtual Queue *get_partition_queue (TopicPartition *partition) = 0;
```

Comment:
expose on Consumer as well, or you could put this on the base Handle and error out if Handle type is Producer